### PR TITLE
Enumerated Collision tuple access

### PR DESCRIPTION
**Pull request by TelloViz**
I added an enum to the private section of the game.h file. 
I did this to increase readability in places where the collision tuple was being used.

### CODE REVIEW

File: Breakout/src/game.h [hunk 1]
```diff
@@ -41,6 +41,7 @@ class Game
 	void processInput(float delta);
 	bool checkCollision(const sf::RectangleShape& rec1, const sf::RectangleShape& rec2);
 	std::tuple<bool, Direction, sf::Vector2f> checkCollision(const sf::CircleShape& circle, const sf::RectangleShape& rec);
+	enum collisionInfoKey{ HasCollKey, DirKey, PosKey };
```
Comment (zann1x) [inline Breakout/src/game.h]:
```suggestion
	enum CollisionInfo { collisionHappened, collisionDirection, collisionPosition };
```

File: Breakout/src/game.cpp [hunk 2]
```diff
@@ -164,10 +164,10 @@ void Game::update(float delta)
 
 		// check if ball has contact with player paddle
 		std::tuple<bool, Direction, sf::Vector2f> coll = checkCollision(m_ball.getDrawable(), m_paddle.getDrawable());
-		if (std::get<0>(coll))
+		if (std::get<HasCollKey>(coll))
```
Comment (zann1x) [inline Breakout/src/game.cpp]:
```suggestion
		if (std::get<collisionHappened>(coll))
```

File: Breakout/src/game.cpp [hunk 3]
```diff
@@ -188,10 +188,10 @@ void Game::update(float delta)
 		for (size_t i = 0; i < m_objects.size(); )
 		{	
 			std::tuple<bool, Direction, sf::Vector2f> coll = checkCollision(m_ball.getDrawable(), m_objects.at(i).getDrawable());
-			if (std::get<0>(coll))
+			if (std::get<HasCollKey>(coll))
```
Comment (zann1x) [inline Breakout/src/game.cpp]:
```suggestion
			if (std::get<collisionHappened>(coll))
```

File: Breakout/src/game.cpp [hunk 4]
```diff
@@ -164,10 +164,10 @@ void Game::update(float delta)
 
 		// check if ball has contact with player paddle
 		std::tuple<bool, Direction, sf::Vector2f> coll = checkCollision(m_ball.getDrawable(), m_paddle.getDrawable());
-		if (std::get<0>(coll))
+		if (std::get<HasCollKey>(coll))
 		{
-			Direction dir = std::get<1>(coll);
-			const sf::Vector2f& diffVec = std::get<2>(coll);
+			Direction dir = std::get<DirKey>(coll);
```
Comment (zann1x) [inline Breakout/src/game.cpp]:
```suggestion
			Direction dir = std::get<collisionDirection>(coll);
```

File: Breakout/src/game.cpp [hunk 5]
```diff
@@ -164,10 +164,10 @@ void Game::update(float delta)
...
+			const sf::Vector2f& diffVec = std::get<PosKey>(coll);
```
Comment (zann1x) [inline Breakout/src/game.cpp]:
```suggestion
			const sf::Vector2f& diffVec = std::get<collisionPosition>(coll);
```

File: Breakout/src/game.cpp [hunk 6]
```diff
@@ -188,10 +188,10 @@ void Game::update(float delta)
 		for (size_t i = 0; i < m_objects.size(); )
 		{	
 			std::tuple<bool, Direction, sf::Vector2f> coll = checkCollision(m_ball.getDrawable(), m_objects.at(i).getDrawable());
-			if (std::get<0>(coll))
+			if (std::get<HasCollKey>(coll))
 			{
-				Direction dir = std::get<1>(coll);
-				const sf::Vector2f& diffVec = std::get<2>(coll);
+				Direction dir = std::get<DirKey>(coll);
```
Comment (zann1x) [inline Breakout/src/game.cpp]:
```suggestion
				Direction dir = std::get<collisionDirection>(coll);
```

File: Breakout/src/game.cpp [hunk 7]
```diff
@@ -188,10 +188,10 @@ void Game::update(float delta)
...
+				const sf::Vector2f& diffVec = std::get<PosKey>(coll);
```
Comment (zann1x) [inline Breakout/src/game.cpp]:
```suggestion
				const sf::Vector2f& diffVec = std::get<collisionPosition>(coll);
```

**Comment by TelloViz** [comment]
By the way... I hope it's cool that I'm kinda using your Breakout repo as a springboard into git. I can just keep the changes to my own repository if you'd rather not deal with my novice commits? Let me know.

-Josh

**Review by zann1x — CHANGES_REQUESTED** [review]

**Comment by zann1x** [comment]
Oh no it's fine, feel free to keep working on this repository 😄

**Comment by TelloViz** [comment]
Cool, thanks bro. Git Hub is a little daunting at first especially when you've not done a lot of collaborative coding yet. I don't know how many names I switched between for those enums but I'm happy for the corrections you made.

**Review by zann1x — APPROVED** [review]

**Pull request merged**